### PR TITLE
Hum up

### PR DESCRIPTION
**Pull request by Stupple**
only thing error this the norm Archer  some how it done like running with out armour

### CODE REVIEW

File: Client/MirObjects/PlayerObject.cs [hunk 1]
```diff
@@ -531,13 +1054,22 @@ public virtual void SetLibraries()
...
+                    {
+                        WeaponLibrary1 = 0 < Libraries.Fishing.Length ? Libraries.Fishing[Weapon - 49] : null;
+                        WeaponLibrary2 = null;
+                        WeaponOffSet = -632;
+                    }
+                    else
+                    {
+                        WeaponLibrary1 = 0 < (Gender == MirGender.Male ? Libraries.UpFishingM.Length : Libraries.UpFishingF.Length) ? (Gender == MirGender.Male ? Libraries.UpFishingM[Weapon - 49] : Libraries.UpFishingF[Weapon - 49]) : null;
+                        WeaponLibrary2 = null;
+                        WeaponOffSet = -600;
```
Comment (Suprcode) [inline Client/MirObjects/PlayerObject.cs]: you sure this offset is correct? the original one was -632 however now that only applies to highWarrior

File: Client/MirScenes/SelectScene.cs [hunk 2]
```diff
@@ -405,7 +434,7 @@ public void StartGame(S.StartGameBanned p)
             StartGameButton.Enabled = true;
 
             TimeSpan d = p.ExpiryDate - CMain.Now;
-            MirMessageBox.Show(string.Format("This account is banned.\n\nReason: {0}\nExpiryDate: {1}\nDuration: {2:#,##0} Hours, {3} Minutes, {4} Seconds", p.Reason,
+            MirMessageBox.Show(string.Format("This account is banned.\n\nReason: {0}\nExpiaryDate: {1}\nDuration: {2:#,##0} Hours, {3} Minutes, {4} Seconds", p.Reason,
```
Comment (Suprcode) [inline Client/MirScenes/SelectScene.cs]: small typo here for "ExpiaryDate" should be "ExpiryDate"

File: Client/MirScenes/SelectScene.cs [hunk 3]
```diff
@@ -507,7 +551,7 @@ protected override void Dispose(bool disposing)
         #endregion
         public sealed class NewCharacterDialog : MirImageControl
         {
-            private static readonly Regex Reg = new Regex(@"^[A-Za-z0-9]{" + Globals.MinCharacterNameLength + "," + Globals.MaxCharacterNameLength + "}$");
+            private static readonly Regex Reg = new Regex(@"^[A-Za-z0-9ㄱ-힣]{" + Globals.MinCharacterNameLength + "," + Globals.MaxCharacterNameLength + "}$");
```
Comment (Suprcode) [inline Client/MirScenes/SelectScene.cs]: no need for these special characters in the client - should be changed back to "[A-Za-z0-9]"

**Comment by Suprcode** [comment]
I'm missing 3 dialogs when i try to compile the client. have you checked the new files in?

public TransformImageDialog TransformImageDialog; //stupple
        public TransformBackDialog TransformBackDialog;  //stupple
        public HumupTransformDialog HumupTransformDialog; //stupple

**Comment by Stupple** [comment]
public TransformImageDialog TransformImageDialog; //stupple
public TransformBackDialog TransformBackDialog; //stupple
public HumupTransformDialog HumupTransformDialog; //stupple

are in follwing as i dont no how to att i auto so i need to do it manual 
it called  
TransformImageDialog.cs

need to be add in Client.MirScenes.Dialogs

but tbh this npc are not fulled coded yet as it for transform items and i no idea what the Krona is like

File: Client/MirObjects/PlayerObject.cs (hunk 1 above)
Comment (Stupple) [inline Client/MirObjects/PlayerObject.cs]: tbh o not try finished

File: Client/MirScenes/SelectScene.cs (hunk 3 above)
Comment (Stupple) [inline Client/MirScenes/SelectScene.cs]: tbh most select scene had funy feeling that be one need looking at

**Comment by Suprcode** [comment]
if you right click the physical file you should be able to add to commit - its just called "Add..."

**Comment by Stupple** [comment]
that what i did

**Comment by Suprcode** [comment]
found it and added.

i'm finding so many issues with this commit unfortunately.

The items are displaying their class requirement twice (sometimes once in red)
Archers placements are wrong when walking
I get a weird buffer error on NPC goods  when reloading the server (no idea why that should affect it)

that's just when i was logged in for 2 seconds, haven't even tried to use the hum up system yet.

**Comment by Stupple** [comment]
The items are displaying their class requirement twice (sometimes once in red)
 = yes i no about this can take it off. i add it have i some time ie allow you to put item on cus ur class and worryed allso u need to update ur bd to march the right one this all found in 

Common.cs

public enum RequiredClass : byte //Stupple = new class var

 public static bool EqualClass(RequiredClass requiredClass, MirClass job) = 2nd line for class

change this to old bd maybe better dont no

Archers placements are wrong when walking
this is when you not got armour on right? i notic this after i tryed updating on new files. and looking at it sill no idea what happed as other working maybe miss spelled thing

I get a weird buffer error on NPC goods when reloading the server (no idea why that should affect it). this cus of DB issuer i look at this now  should be easy it maybe cus i try to update DB check for class and Magic 4 skills. so you dont need to start new ADB. i do this now anyway

**Comment by Suprcode** [comment]
ok thanks. i'd like to get this update in, but don't want to rush it in with lots of bugs thats all

**Comment by Stupple** [comment]
ye i no i keep round some or some chage i fixed the last bug but not it dont it all over i geting different erro then u witch some to do with npcs i cant use guthub windown tool. i can look it to DB should be easy as i have the erro befor and fixed it i no it DB as soon u open it up it crashed ..

Archers placements are wrong when walking. <<<< this comfier that you can walk with Armour but no naked?

and do u if class requirement twice take offer this should be easy

**Comment by Stupple** [comment]
archer fixed  just look at the DB error this weekend. then hope full try to get the ADB to not reset

**Comment by Stupple** [comment]
u retry this dele the good folder?

**Comment by Suprcode** [comment]
not sure what you mean sorry?

**Comment by Stupple** [comment]
I get a weird buffer error on NPC goods when reloading the server (no idea why that should affect it

this should be fix by delete the good folder?

**Pull request closed without merging**